src/preact/index.js

```diff
@@ -146,3 +147,22 @@ export function useCallback(cb, opt_deps) {
 export function toChildArray(unusedChildren) {
   return preact.toChildArray.apply(undefined, arguments);
 }
+
+/**
+ * @param {function(JsonObject, {current: (T|null)}):PreactDef.Renderable} fn
+ * @return {PreactDef.Renderable}
+ * @template T
+ */
+export function forwardRef(fn) {
+  return compat.forwardRef(fn);
+}
+
+/**
+ * @param {{current: (T|null)}} ref
+ * @param {!Function} init
```

Comment:
I'm unsure about defining the type for `init`. React types it like:

```ts
  function useImperativeHandle<T, R extends T>(
    ref: Ref<T>|undefined,
    init: () => R,
    deps?: DependencyList
  ): void;
```

I'm not familiar with any methods to express `R extends T` from Closure generics.

Reply:
We're using `T` for the instance without the handle, it seems like the set of properties  returned by the imperative handle callback is a different set? Not sure why it would be the same as the undecorated component type.

Reply:
This hook is not necessary for this PR any longer, but will be in the future. I removed it to move along for now.

Reply:
I'm not sure we need to type the `init` return value. It's used by the code that creates the ref, but I'm not sure we could ever teach Closure about that.

Reply:
FYI #29650 introduced `useImperativeHandle` so you can grab it from master. :)